modules/openapi-generator/src/main/resources/C-libcurl/apiClient.c.mustache

```diff
@@ -181,14 +181,20 @@ char *assembleTargetUrl(char    *basePath,
     }
 
     if(queryParameters != NULL) {
+        CURL *curl = curl_easy_init();
         strcat(targetUrl, "?");
         list_ForEach(listEntry, queryParameters) {
             keyValuePair_t *pair = listEntry->data;
             replaceSpaceWithPlus(pair->key);
             strcat(targetUrl, pair->key);
             strcat(targetUrl, "=");
-            replaceSpaceWithPlus(pair->value);
-            strcat(targetUrl, pair->value);
+            if(curl) {
+              char *output = curl_easy_escape(curl, pair->value, 0);
```

Comment:
PATH   : Http://SomeUrl.com:8080//store/curlybraces-%7B%7D/AZaz-09._~!$&'(abc)*+, ;=:@Œë🍇ℬ/order/template-{orderId}/.././c///g?c=3&a=1&b=9&c=0#target

This is how the above URL is supposed to be converted. Note characters like :, /, $ ! are not percent encoded.

ESCAPED: http://someurl.com:8080//store/curlybraces-%7B%7D/AZaz-09._~!$&'(abc)*+,%20;=:@%C5%92%C3%AB%F0%9F%8D%87%E2%84%AC/order/template-{orderId}/.././c///g?c=3&a=1&b=9&c=0#target

Using curl_easy_escape, I get a lot more character escapes:
Http%3A%2F%2FSomeUrl.com%3A8080%2F%2Fstore%2Fcurlybraces-%257B%257D%2FAZaz-09._~%21%24%26%27%28abc%29%2A%2B%2C%20%3B%3D%3A%40%C5%92%C3%AB%F0%9F%8D%87%E2%84%AC%2Forder%2Ftemplate-%7BorderId%7D%2F..%2F.%2Fc%2F%2F%2Fg%3Fc%3D3%26a%3D1%26b%3D9%26c%3D0%23target